wifi sample code fails to build when MINIMAL_LIBC is used

**Describe the bug**
Building wifi sample code with WIFI_SHELL and MINIMAL_LIBC enabled causes build failure: 

```
/home/sylvio/espressif/zephyr/zephyr/subsys/net/l2/wifi/wifi_shell.c:565: undefined reference to `sscanf'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**To Reproduce**
Steps to reproduce the behavior:
1. west build -p -b esp32s3_devkitm/esp32s3/procpu samples/net/wifi -DCONFIG_MINIMAL_LIBC=y

**Expected behavior**
Build to work.

**Impact**
Can´t use MINIMAL_LIBC with wifi_shell as ' sscanf' is not available in MINIMAL_LIBC.

**Environment (please complete the following information):**
 - Linux
 -  Zephyr SDK 0.16.8
 - v3.7.0-rc1

--- a/subsys/net/l2/wifi/wifi_shell.c
+++ b/subsys/net/l2/wifi/wifi_shell.c
@@ -567,10 +567,11 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 			params->mfp = atoi(optarg);
 			break;
 		case 'm':
-			sscanf(optarg, "%hhx:%hhx:%hhx:%hhx:%hhx:%hhx",
-				&params->bssid[0], &params->bssid[1],
-				&params->bssid[2], &params->bssid[3],
-				&params->bssid[4], &params->bssid[5]);
+			if (net_bytes_from_str(params->bssid, sizeof(params->bssid),
+					       optarg) < 0) {
+				PR_WARNING("Invalid MAC address\n");
+				return -EINVAL;
+			}
 			break;
 		case 't':
 			if (iface_mode == WIFI_MODE_INFRA) {

